hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/wal/AsyncFSWAL.java

```diff
@@ -630,11 +630,21 @@ protected long append(RegionInfo hri, WALKeyImpl key, WALEdit edits, boolean inM
 
   @Override
   public void sync() throws IOException {
+    sync(useHsync);
+  }
+
+  @Override
+  public void sync(long txid) throws IOException {
+    sync(txid, useHsync);
+  }
+
+  @Override
+  public void sync(boolean forceSync) throws IOException {
```

Comment:
Is this right? The method is sync. We pass a forceSync flag. There are conditions under which we won't sync when we call sync such that we might need a force?

Reply:
Oh, I see you are just using what was there... moving it up into abstract wal that this wal impl uses. Ok. The doc of the sync types can be afterward.